android/src/main/java/com/tailscale/ipn/ui/localapi/LocalAPIClient.kt

```diff
@@ -6,20 +6,35 @@ package com.tailscale.ipn.ui.localapi
 
 import android.util.Log
 import com.tailscale.ipn.ui.model.*
-import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
 
-// A response from the echo endpoint.
 typealias StatusResponseHandler = (Result<IpnState.Status>) -> Unit
-
 typealias BugReportIdHandler = (Result<BugReportID>) -> Unit
-
 typealias PrefsHandler = (Result<Ipn.Prefs>) -> Unit
 
 class LocalApiClient {
     constructor() {
         Log.d("LocalApiClient", "LocalApiClient created")
     }
 
+    companion object {
```

Comment:
Why make this a companion vs making LocalAPIClient a singleton?

Reply:
Singletons make testing (when we get there) hard.   They way I'm going to bolt this together is to have a manager object that owns a single instance of both the Notifier and the LocalAPIClient and the DataModel.   That manager will be a singleton.  This allows you to inject concrete instances of the various things later.

I may even opt to do away with that, and have each viewModel just have it's own notifier and LocalAPIClient.

Reply:
FYI - I've removed all of the singletons in a future patch.  Actions are injected upstream.  Much cleaner.  There's still a single IPNManager owned by the MainActivity which owns a single Notifier and LocalAPIClient.  None of those references are passed around though.